tasks/gene_typing/task_snippy_variants.wdl

```diff
@@ -50,6 +50,46 @@ task snippy_variants {
 
     # Compress output dir
     tar -cvzf "./~{samplename}_snippy_variants_outdir.tar" "./~{samplename}"
+
+    # compute number of reads aligned to reference
+    samtools view -c "~{samplename}/~{samplename}.bam" > READS_ALIGNED_TO_REFERENCE
+
+    # create coverage stats file
+    samtools coverage "~{samplename}/~{samplename}.bam" -o "~{samplename}/~{samplename}_coverage.tsv"
+
+    # capture number of variants from summary file
+    grep "VariantTotal" ~{samplename}/~{samplename}.txt | cut -f 2 > VARIANTS_TOTAL
+
+    # # compute proportion of reference genome with depth >= min_coverage
+    # compute read depth at every position in the genome
+    samtools depth -a "~{samplename}/~{samplename}.bam" -o "~{samplename}/~{samplename}_depth.tsv"
+
+    # if min_coverage is provided, use that value, otherwise use 10
+    if [[ ! -z "~{min_coverage}" ]]; then
+      min_cov="~{min_coverage}"
+    else
+      min_cov=10
+    fi
+
+    # compute reference genome length
+    reference_length=$(cat "~{samplename}/~{samplename}_depth.tsv" | wc -l)
+    echo $reference_length | tee REFERENCE_LENGTH
+
+    # filter depth file to only include positions with depth >= min_cov
+    awk -F "\t" -v cov_var=$min_cov '{ if($3 >= cov_var) {print }}' "~{samplename}/~{samplename}_depth.tsv" > "~{samplename}/~{samplename}_depth_${min_cov}.tsv"
+    
+    # compute proportion of genome with depth >= min_cov
+    reference_length_passed_depth=$(cat "~{samplename}/~{samplename}_depth_${min_cov}.tsv" | wc -l)
+    echo $reference_length_passed_depth | tee REFERENCE_LENGTH_PASSED_DEPTH
```

Comment:
You can also pipe all three of these commands together: 
`samtools depth -a "~{samplename}/~{samplename}.bam" | awk -F '\t' '{if ($3 >= cov_var) print;}' | wc -l`